pkg/operator/ceph/file/status.go

```diff
@@ -28,20 +28,22 @@ import (
 )
 
 // updateStatus updates a fs CR with the given status
-func (r *ReconcileCephFilesystem) updateStatus(observedGeneration int64, namespacedName types.NamespacedName, status cephv1.ConditionType, info map[string]string) {
+func (r *ReconcileCephFilesystem) updateStatus(observedGeneration int64, namespacedName types.NamespacedName, status cephv1.ConditionType, info map[string]string) error {
```

Comment:
Instead of returning an error, another idea is to just add a retry loop to try three times to get the latest and update the status.